modules/system/assets/js/framework.js

```diff
@@ -29,6 +30,14 @@ if (window.jQuery.request !== undefined) {
             throw new Error('Invalid handler name. The correct handler name format is: "onEvent".')
         }
 
+        if (document.cookie) {
+            // Retrieve CSRF cookie
+            var exists = new RegExp("(?:^|;\\s*)" + encodeURIComponent('csrfToken').replace(/[\-\.\+\*]/g, "\\$&") + "\\s*\\=")
+            if (exists.test(document.cookie)) {
+                csrfToken = decodeURIComponent(document.cookie.replace(new RegExp("(?:(?:^|.*;)\\s*" + encodeURIComponent('csrfToken').replace(/[\-\.\+\*]/g, "\\$&") + "\\s*\\=\\s*([^;]*).*$)|^.*$"), "$1")) || null
```

Comment:
What's the complex regex doing?

Reply:
@LukeTowers that checks and extracts the cookie in pure JS. The code is based on this MDN document: https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie/Simple_document.cookie_framework

Reply:
Does jQuery have something built in that's less complicated?

Reply:
It does - jquery.cookie - which the Backend uses. However, I went this route as it was possible people are already including that in their themes, and I didn't want to introduce another hard dependency that may conflict with theme assets.

Reply:
With regards to jquery.cookie it is very old and out of date and has many XSS issues, you can see their release notes for more details. Here is one such example: https://github.com/js-cookie/js-cookie/pull/400

October uses: jQuery Cookie Plugin v1.4.1

Current version is v3.0.0

I will create a pr update for this now.